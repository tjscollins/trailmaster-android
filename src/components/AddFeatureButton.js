/*----------React-----------*/
import React, {PropTypes} from 'react';
import {Text, View, TouchableOpacity} from 'react-native';

/*----------Redux----------*/
import {connect} from 'react-redux';
import * as actions from '../redux/actions';

const AddFeatureButton = (props) => {
  const {styles, text, onPress} = props;
  return (
    <TouchableOpacity onPress={onPress} >
      <View style={styles.featureButtonStyle}>
        <Text style={styles.featureButtonTextStyle}>
          {text}
        </Text>
      </View>
    </TouchableOpacity>
  );
}

AddFeatureButton.propTypes = {
  styles: PropTypes.object,
  text: PropTypes.string,
  onPress: PropTypes.func,
};

export default AddFeatureButton;
