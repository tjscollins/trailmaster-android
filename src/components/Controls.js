/*----------Modules----------*/
import React, {Component} from 'react';
import {Text, View,} from 'react-native';
import axios from 'axios';

class Controls extends Component {
  state = {geoJSON: []}
  componentWillMount() {
    Promise.all([
      axios.get('https://trailmaster.herokuapp.com/pois'),
      axios.get('https://trailmaster.herokuapp.com/routes'),
    ]).then(res => {
      this.setState({geoJSON: [...res[0].data.pois, ...res[1].data.routes]})
    }).catch(err => {
      console.error('Error fetching data', err);
    });
  }
  render() {
    console.log(this.state);
    return (
      <View>
        <Text>Controls Panel</Text>
      </View>
    );
  }
}

export default Controls;
