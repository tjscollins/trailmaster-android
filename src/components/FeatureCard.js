import React from 'react';
import { View } from 'react-native';

const styles = {
  containerStyle: {
    borderBottomWidth: 0,
    borderColor: '#ddd',
    borderRadius: 2,
    borderWidth: 1,

    shadowColor: '#000',
    shadowOffset: {width: 0, height: 2},
    shadowOpacity: 0.1,
    shadowRadius: 2,
    elevation: 1,
    marginLeft: 5,
    marginRight: 5,
    marginTop: 10,
  }
};

const FeatureCard = (props) => {
  return (
    <View style={styles.containerStyle}>
      {props.children}
    </View>
  );
};


export default FeatureCard;
