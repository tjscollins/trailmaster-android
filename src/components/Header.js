/*----------React----------*/
import React, {Component} from 'react';
import {
  Text,
  View,
  TouchableOpacity,
  StyleSheet,
  Modal,
  TouchableHighlight,
  TextInput
} from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';

/*----------Redux----------*/
import {connect} from 'react-redux';
import * as actions from '../redux/actions';

/*----------Style Sheets----------*/
import EStyleSheet from 'react-native-extended-stylesheet';

// const styles = EStyleSheet.create({
//
//   headerStyle: {
//
//   },   searchBoxStyle: {     width: 200,     height: 25,     top: 50, right:
// 0,     left: 100   },
//
// });

class Header extends Component {
  state = {
    // minimizedHeader: true,
    searchVisible: false,
    searchText: '',
    shouldAutoCorrect: false,
    transparentSearch: false
  }
  searchButton() {
    console.log('Pressed Search!');
    this.setState({
      searchVisible: !this.state.searchVisible
    });
  }
  settings() {
    console.log('Pressed Settings!')
  }
  toggleHeader() {
    console.log('Toggle Header Size');
    this
      .props
      .dispatch(actions.toggleHeader());
    // this.setState({minimizedHeader: !this.state.minimizedHeader});
  }
  render() {
    console.log('Props passed to Header: ', this.props);
    const {headerText, UI} = this.props;
    const styles = EStyleSheet.create({
      headerStyle: {
        width: UI.minimizedHeader
          ? 150
          : undefined,
        alignItems: 'center',
        backgroundColor: '#f2f2f2',
        elevation: 2,
        flexDirection: 'row',
        height: 50,
        justifyContent: 'space-between',
        position: 'relative',
        borderBottomRightRadius: UI.minimizedHeader
          ? 5
          : 0
      },
      titleStyle: {
        fontSize: 20,
        paddingLeft: 15,
        // width: '50%',
      },
      toggleButtonStyle: {
        marginRight: 10
      },
      searchButtonStyle: {
        marginRight: 10,
        color: 'steelblue',
        opacity: UI.minimizedHeader
          ? 0
          : 1
      },
      mapButtonStyle: {
        marginRight: 10,
        color: 'olivedrab',
        opacity: UI.minimizedHeader
          ? 0
          : 1
      },
      settingsButtonStyle: {
        marginRight: 15,
        opacity: UI.minimizedHeader
          ? 0
          : 1,
        color: '#aaa'
      }
    });
    return (
      <View style={styles.headerStyle}>
        <Text style={styles.titleStyle}>{headerText}</Text>
        {((display) => {
          if (display)
          return (
            <TouchableOpacity
              onPress={this
                .searchButton
                .bind(this)}>
              <Icon style={styles.searchButtonStyle} name='search' size={20}/>
            </TouchableOpacity>
          );
        }
        )(!UI.minimizedHeader)}
        {((display) => {
          if (display)
          return (
            <TouchableOpacity
              onPress={this
                    .searchButton
                .bind(this)}>
              <Icon style={styles.mapButtonStyle} name='map' size={20}/>
            </TouchableOpacity>
          );
        }
        )(!UI.minimizedHeader)}
        {((display) => {
          if(display)
          return (
            <TouchableOpacity onPress={this
                .settings
              .bind(this)}>
              <Icon style={styles.settingsButtonStyle} name='cog' size={20}/>
            </TouchableOpacity>
          )
        })(!UI.minimizedHeader)}
        <TouchableOpacity
          onPress={this
            .toggleHeader
            .bind(this)}>
          <Icon
            style={styles.toggleButtonStyle}
            name={UI.minimizedHeader
              ? 'chevron-right'
            : 'chevron-left'}
            size={20}/>
        </TouchableOpacity>

        {/* <Modal
          animationType={"none"}
          transparent={this.state.transparentSearch}
          visible={this.state.searchVisible}
          onRequestClose={() => {
            alert("Modal has been closed.")
          }}>
          <View style={styles.searchBoxStyle}>
            <View>
              <Text>Hello World!</Text>

              <TouchableHighlight
                onPress={() => this.setState({
                  searchVisible: !this.state.searchVisible
                })}>
                <Text>Hide Modal</Text>
              </TouchableHighlight>
              <TextInput
                autoCorrect={this.state.shouldAutoCorrect}
                onChangeText={(searchText) => this.setState({searchText})} placeholder={'Search'}
              />

            </View>
          </View>
        </Modal> */}
      </View>
    );
  }
}

Header.propTypes = {
  dispatch: React.PropTypes.func,
  UI: React.PropTypes.object,
  headerText: React.PropTypes.string
};

export default connect(state => state)(Header);
