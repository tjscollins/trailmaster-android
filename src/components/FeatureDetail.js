import React from 'react';
import { View, Text } from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';

import FeatureCard from './FeatureCard';
import FeatureSection from './FeatureSection';

const styles = {
  titleViewStyle: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  titleStyle: {
    fontSize: 18,
    fontWeight: '600',
    color: '#333',
  },
  attributeStyle: {
    flexDirection: 'row',
    padding: 5,
  },
  attributeTextStyle: {
    left: 20,
    width: 250,
  }
}

const FeatureDetail = ({feature}) => {
  const {name, desc, condition} = feature.properties;
  const {type} = feature.geometry;
  return (
    <FeatureCard>
      <FeatureSection>
        <View style={styles.titleViewStyle}>
          <View style={{width: 275}}>
            <Text style={styles.titleStyle}>{name}</Text>
          </View>
          <Icon style={{}} name={type === 'Point' ? 'map-marker' : 'road'} size={20}/>
        </View>
      </FeatureSection>

      <FeatureSection>
        <View style={styles.attributeStyle}>
          <View style={{width: 80}}>
            <Text>Description:</Text>
          </View>
          <Text style={styles.attributeTextStyle}>{desc}</Text>
        </View>
        <View style={styles.attributeStyle}>
          <View style={{width: 80}}>
            <Text>Condition:</Text>
          </View>
          <Text style={styles.attributeTextStyle}>{condition}</Text>
        </View>
      </FeatureSection>
    </FeatureCard>
  );
};

export default FeatureDetail;
