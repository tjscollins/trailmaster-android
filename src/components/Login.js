/*----------React----------*/
import React, {Component} from 'react';
import {Text, View, TextInput, TouchableOpacity} from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';

/*----------Redux----------*/
import {connect} from 'react-redux';
import * as actions from '../redux/actions';

/*----------Components----------*/
import Header from './Header';

/*----------Style Sheets----------*/
import EStyleSheet from 'react-native-extended-stylesheet';

class Login extends React.Component {
  render() {
    const styles = EStyleSheet.create({
      containerStyle: {
        flex: 1
      },
      loginFormStyle: {
        margin: 10,
        flex: 1,
        backgroundColor: 'white',
        borderWidth: 1,
        borderColor: '#ddd',
        borderRadius: 2,
        padding: 10,
        paddingTop: '25%',
        flexDirection: 'column',
        alignItems: 'center',
        justifyContent: 'flex-start',
      },
      formRow: {
        flexDirection: 'row',
        alignItems: 'center',
        justifyContent: 'center',
        // borderBottomWidth: 1,
        // borderColor: '#ddd',

      },
      formLabel: {
        width: 75,
        marginBottom: -10,
      },
      formInput: {
        width: 250,
        textAlign: 'center',
      },
      loginButton: {
        margin: 10,
        padding: 10,
        width: 100,
        justifyContent: 'space-around',
        alignItems: 'center',
        borderWidth: 1,
        borderColor: '#ddd',
        borderRadius: 2,
        backgroundColor: '$primaryButtonColor'
      }
    });
    return (
      <View style={styles.containerStyle}>
        <View style={styles.loginFormStyle}>
          <View style={styles.formRow}>
            <Text style={styles.formLabel}>Email: </Text>
            <TextInput autoFocus keyboardType={'email-address'} placeholder={'Email Address'} style={styles.formInput}/>
          </View>
          <View style={styles.formRow}>
            <Text style={styles.formLabel}>Password: </Text>
            <TextInput secureTextEntry autoCorrect={false} placeholder={'Password'} style={styles.formInput}/>
          </View>
          <View style={styles.formRow}>
            <Text style={styles.formLabel}></Text>
            <TouchableOpacity style={styles.loginButton}>
              <Text>
                Login
              </Text>
            </TouchableOpacity>
          </View>
        </View>
      </View>
    );
  }
}

export default connect(state => state)(Login);
