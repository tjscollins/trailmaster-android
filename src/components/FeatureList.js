/*----------Modules----------*/
import React, {Component} from 'react';
import {ScrollView} from 'react-native';
import {connect} from 'react-redux';
import EStyleSheet from 'react-native-extended-stylesheet';

/*----------Components----------*/
import FeatureDetail from './FeatureDetail';

/*----------Redux----------*/
// import * as actions from '../redux/actions';

/*----------API----------*/
// import {fetchData, validateServerData} from '../api/TrailmasterAPI';

class FeatureList extends Component {
  renderGeoJSON() {
    return this
      .props
      .geoJSON
      .features
      .map((feature) => <FeatureDetail key={feature._id + 'geoJSON-list'} feature={feature}/>);
  }
  render() {
    const styles = EStyleSheet.create({
      scrollViewStyle: {},
    });
    return (
      <ScrollView style={styles.scrollViewStyle}>
        {this.renderGeoJSON()}
      </ScrollView>
    );
  }
}



FeatureList.propTypes = {
  // dispatch: React.PropTypes.func,
  // userSession: React.PropTypes.object,
  geoJSON: React.PropTypes.object,
};

export default connect(state => state)(FeatureList);
