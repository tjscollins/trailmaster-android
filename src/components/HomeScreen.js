/*----------React----------*/
import React, {Component} from 'react';
import {Text, View, TouchableOpacity} from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';

/*----------Redux----------*/
import {connect} from 'react-redux';
import * as actions from '../redux/actions';

/*----------Components----------*/
import Header from './Header';
import Login from './Login';

/*----------Style Sheets----------*/
import EStyleSheet from 'react-native-extended-stylesheet';

class HomeScreen extends React.Component {
  constructor() {
    super();
  }
  showLogin() {
    this.props.toRoute({
      name: 'Login',
      component: Login,
      statusBarProps: {
        hidden: true
      },
    });
  }
  loginText() {
    const {email} = this.props.userSession;
    if(email) {
      return `Logged in as: ${email}`;
    } else {
      return 'Login';
    }
  }
  myTrailsText() {
    return 'My Trails Button';
  }
  savedMapsText() {
    return 'Saved Maps Button';
  }
  render() {
    const {xAuth} = this.props.userSession;
    const styles = EStyleSheet.create({
      buttonStyle: {
        aspectRatio: 2,
        backgroundColor: 'white',
        borderColor: '#ddd',
        borderRadius: 2,
        borderWidth: 1,
        elevation: 1,
        flex: 1,
        marginTop: 15,
        marginLeft: 30,
        marginRight: 0,
        maxHeight: '$homeScreenButtonHeight',
      },
      buttonTextStyle: {

      },
      // loginStyle: {
      //   flex: 1,
      //   backgroundColor: 'white',
      //   aspectRatio: 2,
      //   borderColor: '#ddd',
      //   borderRadius: 2,
      //   borderWidth: 1,
      //   elevation: 1,
      //   maxHeight: '$homeScreenButtonHeight',
      // },
      // myTrailsStyle: {
      //
      // },
      // savedMapsStyle: {
      //   backgroundColor: 'white',
      //   aspectRatio: 2,
      //   height: 100
      // },
      containerStyle: {
        flex: 1,
        alignItems: 'center',
        justifyContent: 'center',
        flexDirection: 'row',
        flexWrap: 'wrap'
      }
    });
    return (
      <View style={{flex: 1}}>
        <Header toRoute={this.props.replaceRoute} headerText={'Trailmaster'}/>
        <View style={styles.containerStyle}>
          {((isLoggedIn) => {
            if(isLoggedIn) {
              return (
                <TouchableOpacity>
                  <View style={styles.buttonStyle}>
                    <Text>
                      {this.myTrailsText()}
                    </Text>
                  </View>
                </TouchableOpacity>
              );
            }
          })(xAuth)}
          {((isLoggedIn) => {
            if(isLoggedIn) {
              return (
                <TouchableOpacity>
                  <View style={styles.buttonStyle}>
                    <Text>
                      {this.savedMapsText()}
                    </Text>
                  </View>
                </TouchableOpacity>
              );
            }
          })(xAuth)}
          <TouchableOpacity onPress={this.showLogin.bind(this)}>
            <View style={styles.buttonStyle}>
              <Text style={styles.buttonTextStyle}>
                {this.loginText()}
              </Text>
            </View>
          </TouchableOpacity>
        </View>
      </View>
    );
  }
}

HomeScreen.propTypes = {
  userSession: React.PropTypes.object.isRequired,
  toRoute: React.PropTypes.func.isRequired,
  replaceRoute: React.PropTypes.func.isRequired,
};

export default connect(state => state)(HomeScreen);
