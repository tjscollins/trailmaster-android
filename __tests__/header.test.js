/*global it expect*/
import 'react-native';
import React from 'react';
import {Header} from '../src/components/Header';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('should render the headerText prop', () => {
  const headerProps = {
    dispatch: () => {},
    UI: {},
    headerText: 'Test',
    toRoute: () => {},
  };
  const header = renderer.create(<Header {...headerProps} />).toJSON();
  expect(header).toMatchSnapshot();
});
