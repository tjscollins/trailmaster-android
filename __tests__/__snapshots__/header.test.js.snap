exports[`test should render the headerText prop 1`] = `
<View
  style={undefined}>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={undefined}>
    Test
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}>
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 20,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }>
      
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}>
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 20,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }>
      
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}>
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 20,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }>
      
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}>
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 20,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }>
      
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}>
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 20,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }>
      
    </Text>
  </View>
</View>
`;
